rollouts/controllers/rollout_controller.go

```diff
@@ -276,6 +277,14 @@ func (r *RolloutReconciler) getPackageDiscoveryClient(rolloutNamespacedName type
 func (r *RolloutReconciler) reconcileRollout(ctx context.Context, rollout *gitopsv1alpha1.Rollout, strategy *gitopsv1alpha1.ProgressiveRolloutStrategy, packageDiscoveryClient *packagediscovery.PackageDiscovery) error {
 	logger := klog.FromContext(ctx)
 
+	if rollout != nil && rollout.Spec.SyncTemplate != nil {
+		if rollout.Spec.SyncTemplate.Type == gitopsv1alpha1.TemplateTypeRepoSync {
+			err := fmt.Errorf("reposync is not yet supported")
```

Comment:
just a note for us: we will have to start capturing these as errors as conditions in the status part of the API to surface it to the end users. Today, one has to look at the error logs of the controller :)

Reply:
Filed https://github.com/GoogleContainerTools/kpt/issues/3839